### PR TITLE
Revert Fast List(Nested)

### LINKED ISSUE #1424: List(Pluck()) raises while Pluck(many=True) works
For completeness, below is my example where I noticed this but a PR with at least a test added is on it's way and I'll see if I can figure out how to fix it.  This was broken by https://github.com/marshmallow-code/marshmallow/commit/a6fd2b6f759c1b99275f949b7663295f85cc7f27.

https://repl.it/@altendky/UnluckyInfiniteUnits-5
```python3
import attr
import marshmallow


@attr.s
class Member:
    name = attr.ib()
    other = attr.ib()


class MemberSchema(marshmallow.Schema):
    name = marshmallow.fields.String()
    other = marshmallow.fields.String()


@attr.s
class Holder:
    members = attr.ib()

class HolderSchema(marshmallow.Schema):
    members = marshmallow.fields.List(marshmallow.fields.Pluck(MemberSchema, field_name='name'))
    # members = marshmallow.fields.Pluck(MemberSchema, field_name='name', many=True)


def main():
    my_holder = Holder(
        members=[
            Member(name='name_0', other='other_0'),
            Member(name='name_1', other='other_1'),
        ],
    )

    dumped = HolderSchema().dump(my_holder)
    print(dumped)

main()
```
```python-traceback
Traceback (most recent call last):
  File "main.py", line 45, in <module>
    main()
  File "main.py", line 42, in main
    dumped = HolderSchema().dump(my_holder)
  File "/home/runner/.local/share/virtualenvs/python3/lib/python3.7/site-packages/marshmallow/schema.py", line 553, in dump
    result = self._serialize(processed_obj, many=many)
  File "/home/runner/.local/share/virtualenvs/python3/lib/python3.7/site-packages/marshmallow/schema.py", line 517, in _serialize
    value = field_obj.serialize(attr_name, obj, accessor=self.get_attribute)
  File "/home/runner/.local/share/virtualenvs/python3/lib/python3.7/site-packages/marshmallow/fields.py", line 325, in serialize
    return self._serialize(value, attr, obj, **kwargs)
  File "/home/runner/.local/share/virtualenvs/python3/lib/python3.7/site-packages/marshmallow/fields.py", line 695, in _serialize
    return self.inner._serialize(value, attr, obj, many=True, **kwargs)
  File "/home/runner/.local/share/virtualenvs/python3/lib/python3.7/site-packages/marshmallow/fields.py", line 635, in _serialize
    return ret[self._field_data_key]
TypeError: list indices must be integers or slices, not str
```
`requirements.txt`
```
attrs==19.2.0
marshmallow==3.2.1
```

### LINKED ISSUE #1497: Regression on list of nullable nested fields
Hello,

While porting some code from 2.20 to 3.3, I stumbled upon this change in behavior regarding list of nullable fields.

in 2.20 the following code works:
```python
from marshmallow import fields, Schema

class S(Schema):
    f = fields.List(fields.Nested(Schema, allow_none=True))

assert S().dump(dict(f=[None])).data == dict(f=[None])
```
whereas the same code in 3.3 fails with
`AssertionError: assert {'f': [{}]} == {'f': [None]}`
```python
from marshmallow import fields, Schema

class S(Schema):
    f = fields.List(fields.Nested(Schema, allow_none=True))

assert S().dump(dict(f=[None])) == dict(f=[None])
```

I could not find anything related to this in the upgrade instructions or the documentation.
I'm fully aware that this is an edge case and I don't mind looking into it if it's actually considered as a bug.

here are two unit tests that can be used to reproduce this issue:
```python
def test_list_of_nested_nullable_serialize_none_to_none(self):
    res = fields.List(fields.Nested(Schema, allow_none=True)).serialize("foo", {"foo": [None]})
    assert res == [None]

def test_nested_multiple_nullable_serialize_none_to_none(self):
    res = fields.Nested(Schema, allow_none=True, many=True).serialize("foo", {"foo": [None]})
    assert res == [None]
```

### PR DESCRIPTION
**Pull request by deckar01**
`Nested(many)` does not handle `allow_none` properly due to using `Schema.load(many)`. Optimizing `List(Nested)` by reusing this code caused it to inherit this behavior, which was a regression. A regression test was added to avoid this when optimizing `List(Nested)` in the future.

Fixes #1497
Fixes #1424

Edit by @lafrech: Reverts #1306

### CODE REVIEW

**Comment by sloria** [comment]
Sorry, I haven't been following the related issues. On a cursory glance this seems like the right course, but it's unfortunate that it will introduce a performance regression. I defer to your and @lafrech 's judgment for merging and releasing this.

File: src/marshmallow/fields.py [hunk 1]
```diff
@@ -562,33 +562,33 @@ def _nested_normalized_option(self, option_name: str) -> typing.List[str]:
...
         schema = self.schema
         if nested_obj is None:
             return None
-        many = schema.many or self.many or many
-        return schema.dump(nested_obj, many=self.many or many)
+        many = schema.many or self.many
+        try:
+            return schema.dump(nested_obj, many=many)
+        except ValidationError as exc:
+            raise ValidationError(exc.messages, valid_data=exc.valid_data) from exc
```
Comment (lafrech) [inline src/marshmallow/fields.py]: I don't understand the need for this try/except block.

**Review by lafrech — APPROVED** [review]
This perf optim has issues. Reverting it is the right choice, functionaly.

Of course, it would be fantastic if we managed to minimize load/dump time another way. But for now, let's solve the functional issues.

Just one question, but otherwise, approved.

Approved

File: src/marshmallow/fields.py (hunk 1 above)
Comment (deckar01) [reply src/marshmallow/fields.py]: I don't remember having to manually resolve a merge conflict for this line. It wasn't in the original diff, but it was part of the upstream merge that was performed to resolve merge conflicts. I only reverted the 2 commits I added, but it looks like git reverted b922a7c063 also.

I will fix this.

**Comment by lafrech** [comment]
Great.

Should we release as bugfix or minor?

This is a bugfix, but it has a perf impact on the List(Nested()) case.

**Comment by lafrech** [comment]
I'm tempted to label this *not* a breaking change because there is no functional drawback. However, we could make it a minor (not patch) release because of the performance regression.

CHANGELOG proposal:

```
3.5.0 (2020-02-18)
******************

Bug fixes:

- Fix list of nullable nested fields ``List(Nested(Field, allow_none=True)`` (:issue:`1497`).
  Because this fix reverts an optimization introduced to speed-up serialization and
  deserialization of lists of nested fields, a negative impact on performance in this
  specific case is expected.
```

@sloria @deckar01

**Comment by sloria** [comment]
Yep, sounds good to me
